make-typed-request.js

```diff
@@ -36,7 +41,8 @@ function makeTypedRequest(treq, opts, cb) {
             httpVersion: resp.httpVersion,
             headers: resp.headers,
             statusCode: resp.statusCode,
-            body: resp.body
+            body: resp.body,
+            requestUrl: reqOpts.url
```

Comment:
Looks like we are exposing this to make it testable.

Let's write a unit test instead; where we pass in a custom `reqOpts.request`

See https://github.com/uber/typed-request-client/blob/master/test/typed-request-client.js#L54 for an example of this style of test.